packages/flutter_tools/test/commands.shard/hermetic/project_validator_test.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:args/command_runner.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/commands/validate_project.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:flutter_tools/src/project_validator.dart';
+import 'package:flutter_tools/src/project_validator_result.dart';
+
+import '../../src/context.dart';
+import '../../src/test_flutter_command_runner.dart';
+
+class ProjectValidatorDummy extends ProjectValidator {
+  @override
+  Future<List<ProjectValidatorResult>> start(FlutterProject project) async{
+    return <ProjectValidatorResult>[
+      const ProjectValidatorResult(name: 'pass', value: 'value', status: StatusProjectValidator.success),
+      const ProjectValidatorResult(name: 'fail', value: 'my error', status: StatusProjectValidator.error),
+      const ProjectValidatorResult(name: 'pass two', value: 'pass', warning: 'my warning', status: StatusProjectValidator.warning),
+    ];
+  }
+
+  @override
+  bool supportsProject(FlutterProject project) {
+    return true;
+  }
+}
+
+class ProjectValidatorSecondDummy extends ProjectValidator {
+  @override
+  Future<List<ProjectValidatorResult>> start(FlutterProject project) async{
+    return <ProjectValidatorResult>[
+      const ProjectValidatorResult(name: 'second', value: 'pass', status: StatusProjectValidator.success),
+      const ProjectValidatorResult(name: 'second fail', value: 'second fail', status: StatusProjectValidator.error),
+    ];
+  }
+
+  @override
+  bool supportsProject(FlutterProject project) {
+    return true;
+  }
+}
+
+class ProjectValidatorCrash extends ProjectValidator {
+  @override
+  Future<List<ProjectValidatorResult>> start(FlutterProject project) async{
+    throw Exception('my exception');
+  }
+
+  @override
+  bool supportsProject(FlutterProject project) {
+    return true;
+  }
+}
+
+void main() {
+  FileSystem fileSystem;
+
+  group('analyze project command', () {
+
+    setUp(() {
+      fileSystem = MemoryFileSystem.test();
+    });
+
+    testUsingContext('success, error and warning', () async {
+      final BufferLogger loggerTest = BufferLogger.test();
+      final ValidateProjectCommand command = ValidateProjectCommand(
+          fileSystem: fileSystem,
+          logger: loggerTest,
+          allProjectValidators: <ProjectValidator>[
+            ProjectValidatorDummy(),
+            ProjectValidatorSecondDummy()
+          ]
+      );
+      final CommandRunner<void> runner = createTestCommandRunner(command);
+
+      await runner.run(<String>['validate-project']);
+
+      const String expected = '\n'
+          '┌──────────────────────────────────────────┐\n'
```

Comment:
I think we should differentiate which of these came from which validator, the way we do with the doctor. If that information is not important, then we might as well flatten the structure and make each of these their own validator, and not have `start()` return multiple results.

Reply:
> I think we should differentiate which of these came from which validator, the way we do with the doctor. If that information is not important, then we might as well flatten the structure and make each of these their own validator, and not have `start()` return multiple results.

The main advantage of having more than one result returned was to avoid doing repetitive work on different validators (eg: parse a file more than once). Knowing the parent validator of each result can be beneficial for contributors trying to debug an error on a specific validator but I don't think It could be beneficial for the end user with the current implementation. I think a good fix would be adding a new field on the validators like `title` and first output that and then all of its results. eg:
`
class ProjectValidatorSecondDummy extends ProjectValidator  {
   final String title = 'Android'
}

class ProjectValidatorDummy extends ProjectValidator  {
   final String title = 'IOS tool'
}

//output
const String expected = '\n'
          '┌───────────────────┐\n'
            '│ Android                                                          │\n'
            '│ [✓] pass: value                                             │\n'
            '│ [✗] Error: my error                                       │\n'
            '│ [!] pass two: pass (warning: my warning) │\n'
            '│                                                                       │\n'
            '│ IOS tool                                                        │\n'
            '│ [✓] second: pass                                        │\n'
            '│ [✗] Error: second fail                                  │\n'
          '└──────────────────┘\n';
`

What do you think? @christopherfujino

Reply:
This looks good to me.

Although to be consistent, I think we should always put the task name first (it looks like when there is an error, we print `"Error"` first).

Reply:
done @christopherfujino